src/app/business/routes/model/sagas/index.js

```diff
@@ -36,8 +37,9 @@ function* fetchList(request) {
 
 function* fetchDetail({payload}) {
     const state = yield select();
+    const modelDetailList = itemResults(state, 'model');
 
-    const item = state.model.item.results.find(o => o.traintuple.key === payload.traintuple.key);
+    const item = modelDetailList.find(o => o.traintuple.key === payload.traintuple.key);
```

Comment:
A light in my head says `WARNING`.
I don't remember why yet, but I had a lot of troubles when using selector in sagas code.
Contexts were not the same.
Hope it will remind me.

If we can avoid using selector in sagas, it should be better

Reply:
After reading your comment I went to the documentation and apparently, redux-sagas offers a standard way to use selectors! See https://redux-saga.js.org/docs/api/ (look for select). I'll try to use it, it should be pretty safe.

The alternative was to duplicate the logic of the selector inside the saga, it didn't felt right at all.